### PR TITLE
Prep for service rewrite

### PR DESCRIPTION
**Pull request by DanielElisenberg**
# PREP FOR SERVICE REWRITE

Dataset-builder will be rewritten to be a persistent microservice that polls the job_service for new jobs to run. This PR is mainly for restructuring the app to accomodate the upcoming changes.

## What's changed?
* Moved the add_dataset.py to the service folder as it is now a service. Renamed to build_dataset_service.py to match the name of the service, and because it can result in both ADD and CHANGE_DATA.
* The csv converter and parquet converter functionality felt unnaturally bundled in the same module at this point. The main function of the csv converter is to enrich the csv data with additional columns. Renamed this step "dataset_enricher" and kept the module "dataset_converter" with only the functionality that pertains to conversion to parquet format. This made both modules and tests more readable.
* Removed logging to file as this service must now be configured to log to the ELK stack
* Removed unecessary files
    - conversion_only script: This should be used from migration_branch
    - DatasetMetadataSchema: This is now part of the validator
    - Test resources belonging to file logging
    - MetadataAdapter: this service no longer communicates with metadata-service
* Restructured test resources and files to accomodate changes
* Empty adapter module to be implemented for job-service
* Initial app.py file to be implemented

### What's next?
* Implement job_service connection and persistence
* Implement request to version-service for import built file as DRAFT
* Implement multiprocessing
* Implement logging to ELK

### CODE REVIEW

File: dataset_builder/service/dataset_builder_service.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
+import glob
```
Comment (pawbu) [inline dataset_builder/service/dataset_builder_service.py:1]: Would be nice to follow the hexagonal vocabulary and put this script under` /domain/dataset_builder.py`

File: dataset_builder/service/dataset_builder_service.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        config = config_provider.get_config()
+        working_dir = config['WORKING_DIR']
+        output_dir = config['DATA_OUTPUT_ROOT_DIR']
+        input_dir = config['DATA_INPUT_ROOT_DIR']
+
+        log_config.setup_logger_for_import_pipeline()
+        logger.info(f"Starting to build dataset {dataset_name}")
+
+        # update_job_status("Validating", job_id)
+        __run_validator(dataset_name, input_dir, working_dir)
```
Comment (pawbu) [inline dataset_builder/service/dataset_builder_service.py:32]: Double leading underscore should be unsed when naming class atrributes, so single underscore here and other functions in this project.
Ref: https://peps.python.org/pep-0008/#descriptive-naming-styles

File: dataset_builder/steps/dataset_enricher.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    if date in empty_values:
+        date_value = ''
+        date_epoch_days_value = ''
+    else:
+        date_value = date
+        date_epoch_days_value = days_since_epoch_for[date]
+    return date_value, date_epoch_days_value
+
+
+def enrich_csv(input_csv_path: str, temporal_coverage: dict,
```
Comment (pawbu) [inline dataset_builder/steps/dataset_enricher.py:41]: 💯

File: dataset_builder/steps/dataset_transformer.py [hunk 4]
```diff
@@ -242,10 +241,15 @@ def __get_temporal_coverage(metadata: dict) -> dict:
     return period
 
 
-def transform_metadata(metadata_file_path: str) -> dict:
+def transform_metadata(metadata_file_path: str) -> str:
+    """
```
Comment (pawbu) [inline dataset_builder/steps/dataset_transformer.py:245]: 👍

File: tests/unit/common/test_config_provider.py [hunk 5]
```diff
@@ -2,32 +2,25 @@
 from dataset_builder.common import config_provider
 
 
+ENVIRONMENT = {
```
Comment (pawbu) [inline tests/unit/common/test_config_provider.py:5]: Maybe use pytest-dotenv so that we have the same setup across projects?

**Review by pawbu — APPROVED** [review]
Looks good. Some minor comments.

File: dataset_builder/service/dataset_builder_service.py (hunk 2 above)
Comment (DanielElisenberg) [reply dataset_builder/service/dataset_builder_service.py:32]: True :100: will fix

File: tests/unit/common/test_config_provider.py (hunk 5 above)
Comment (DanielElisenberg) [reply tests/unit/common/test_config_provider.py:5]: Absolutely agree 🏃🏻  I'll fix

File: dataset_builder/service/dataset_builder_service.py (hunk 1 above)
Comment (DanielElisenberg) [reply dataset_builder/service/dataset_builder_service.py:1]: feels less descriptive to me, but it is consistent and not a big deal so I'll rename 👍🏻

**Pull request merged**